src/Compilers/Core/Portable/CommandLine/CommonCompiler.CompilerEmitStreamProvider.cs

```diff
@@ -81,7 +79,7 @@ public override Stream CreateStream(DiagnosticBag diagnostics)
                             else if (e.HResult == eWin32SharingViolation)
                             {
                                 // On Windows File.Delete only marks the file for deletion, but doesn't remove it from the directory.
-                                var newFilePath = Path.Combine(Path.GetDirectoryName(_filePath), Guid.NewGuid().ToString() + "_" + Path.GetFileName(_filePath));
+                                var newFilePath = Path.Combine(Path.GetDirectoryName(_filePath)!, Guid.NewGuid().ToString() + "_" + Path.GetFileName(_filePath));
```

Comment:
>Path.GetDirectoryName(_filePath)! [](start = 63, length = 33)

Should this be an assertion?

Reply:
There is only one case where `GetDirectoryName` returns `null`: when you pass in a directory root. That means when
 passing a file path it will never return `null`. Hence we've generally taken the approach of using suppressions for this API vs. assertions.